nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RetryCount.java

```diff
@@ -0,0 +1,210 @@
+
+/**
+ * Created by jpercivall on 3/17/17.
+ */
+@SupportsBatching
+@SideEffectFree
+@InputRequirement(INPUT_REQUIRED)
+@Tags({"penalize", "retry", "penalize"})
+@CapabilityDescription("Used to facilitate retrying a FlowFile a set number of times before considering the FlowFile 'failed'. Can also be used as a utility to penalize FlowFiles.")
+@WritesAttribute(attribute = RETRY_COUNT_ATTRIBUTE_KEY, description = "The number of times this FlowFile has been routed to 'Retry'.")
+@ReadsAttribute(attribute = RETRY_COUNT_ATTRIBUTE_KEY, description = "The number of times this FlowFile has been routed to 'Retry'.")
+public class RetryCount extends AbstractProcessor {
+
+    public static final String RETRY_COUNT_ATTRIBUTE_KEY = "retryCountProcessor.timesRetried";
+
+    public static final PropertyDescriptor PROP_RETRY_LIMIT = new PropertyDescriptor.Builder()
+            .name("Retry limit")
+            .displayName("Retry limit")
+            .description("The number of times to retry the FlowFile before considering it 'failed' and routing it to 'over limit'.")
+            .expressionLanguageSupported(VARIABLE_REGISTRY)
+            .required(true)
+            .addValidator(StandardValidators.INTEGER_VALIDATOR)
+            .defaultValue("3")
+            .build();
+
+    public static final PropertyDescriptor PROP_PENALIZE_FLOWFILE = new PropertyDescriptor.Builder()
+            .name("Penalize FlowFile")
+            .displayName("Penalize FlowFile")
+            .description("If true then the FlowFiles routed to 'retry' will be penalized.")
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .build();
+
+    public static final PropertyDescriptor PROP_WARN_ON_OVER_LIMIT = new PropertyDescriptor.Builder()
+            .name("Warn on 'over limit'")
+            .displayName("Warn on 'over limit'")
+            .description("If true then when a FlowFile is routed to 'over limit' a message will be logged at the level 'warn'.")
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .build();
+
+    public static final PropertyDescriptor PROP_INPUT_QUEUE_DATA_SIZE_LIMIT = new PropertyDescriptor.Builder()
+            .name("Input queue data size limit")
+            .displayName("Input queue data size limit")
+            .description("If the total input queue data size (a combination of all connections feeding this processor) is above this configured limit then any processed FlowFile is routed to " +
+                    "'over limit'. This is necessary to avoid a back-pressure deadlock caused by with circular back-pressure. The default values are set to be half of what the default " +
+                    "configuration values are for connections back-pressure.")
+            .addValidator(StandardValidators.DATA_SIZE_VALIDATOR)
+            .expressionLanguageSupported(VARIABLE_REGISTRY)
+            .defaultValue("500 mb")
```

Comment:
What about making this not have a default value, since it's not required anyways. I think leaving this feature off will be kind of a default use case (at least in my experience it's what we commonly do). Then users can turn it on if they are prepared to handle "over limit" files.